src/functions/__tests__/typedEnum.test.ts

```diff
@@ -0,0 +1,133 @@
+          enum: [1, 2, 3],
+        };
+
+        expect(() => runTypedEnum(schema, threshold)).toThrowError(AssertionError);
+      },
+    );
+  });
+
+  test('should return undefined when the enum contains no value', () => {
+    const schema = {
+      type: 'integer',
+      enum: [],
+    };
+
+    expect(runTypedEnum(schema, defaultReportingThreshold)).toBeUndefined();
+  });
+
+  describe('basic', () => {
+    test('should return undefined when all enum values respect the format', () => {
+      const schema = {
+        type: 'integer',
+        enum: [123, 456],
+      };
+
+      expect(runTypedEnum(schema, defaultReportingThreshold)).toBeUndefined();
+    });
+
+    test('should identify enum values that do not respect the format', () => {
+      const schema = {
+        type: 'integer',
+        enum: [123, 'a string!', 456, 'and another one!'],
+      };
+
+      expect(runTypedEnum(schema, defaultReportingThreshold)).toEqual([
+        {
+          message: 'Some enum values do not respect the specified type "integer": "a string!", "and another one!".',
+        },
+      ]);
+    });
+  });
+
+  describe('types', () => {
+    const testCases: Array<[string, unknown[], unknown]> = [
+      ['string', ['Hello', 'world!'], 12],
+      ['number', [-2147483648, 17.13], 'Hello'],
+      ['integer', [-2147483648, 17], 12.3],
+      ['boolean', [true, false], 1],
+      // array ?
+      // object ?
+    ];
+
+    test.each(testCases)(
+      'does not report anything when all the definitions are valid for type "%s"',
+      async (type: string, valids: unknown[], invalid: unknown) => {
+        const schema = {
+          type,
+          enum: valids,
+        };
+
+        expect(runTypedEnum(schema, defaultReportingThreshold)).toBeUndefined();
+      },
+    );
+
+    test.each(testCases)(
+      'should identify enum values that do not respect the type "%s"',
+      async (type: string, valids: unknown[], invalid: unknown) => {
+        const schema = {
+          type,
+          enum: [valids[0], invalid],
+        };
+
+        const results = runTypedEnum(schema, defaultReportingThreshold);
+
+        expect(results[0].message).toContain(`: "${invalid}".`);
+      },
+    );
+  });
+
+  describe('reporting threshold', () => {
+    test.each([
+      [1, '"a string!"...'],
+      [2, '"a string!", "and another one!"...'],
+      [3, '"a string!", "and another one!", "seriously?"...'],
+    ])(
+      'should add an ellipsis at the end of the error message when the number of invalid enum values is greater than the reporting threshold "%s"',
+      (threshold, errorMessageTrail) => {
+        const schema = {
+          type: 'integer',
+          enum: ['a string!', 'and another one!', 'seriously?', 'you must be kidding me'],
+        };
+
+        expect(runTypedEnum(schema, threshold)).toEqual([
+          {
+            message: `Some enum values do not respect the specified type "integer": ${errorMessageTrail}`,
```

Comment:
Could the error be reported for each invalid enum value instead of all of them? this might cut out the need for reportingThreshold.

Reply:
@phil Fixed. Multiple messages are now returned and reportingThreahold has been shown the door.